File: C# DLLs/QSenseDotNet/QSenseEventArgs.cs
using System;
using System.Collections.Generic;
using System.Numerics;

namespace QSenseDotNet
{

    /// <summary>
    /// Provides data for the BatteryReceivedEvent
    /// </summary>
    public class BatteryReceivedEventArgs : EventArgs
    {
        internal BatteryReceivedEventArgs(float battery)
        {
            Battery = battery;
        }

        public float Battery { get; set; }
    }

    
    /// <summary>
    /// Provides data for the DeviceNameChangedEvent
    /// </summary>
    public class DeviceNameChangedEventArgs : EventArgs
    {
        internal DeviceNameChangedEventArgs(string name)
        {
            Name = name;
        }

        public string Name { get; set; }
    }

    /// <summary>
    /// Provides data for the EnergyReceivedEvent
    /// </summary>
    public class MotionLevelReceivedEventArgs : EventArgs
    {
        internal MotionLevelReceivedEventArgs(float motionLevel)
        {
            MotionLevel = motionLevel;
        }

        public float MotionLevel { get; set; }
    }

    /// <summary>
    /// Provides data for the StateReceivedEvent
    /// </summary>
    public class StateReceivedEventArgs : EventArgs
    {
        internal StateReceivedEventArgs(bool isOffsetCompensationOn, bool isOffsetCompensated, bool isMagFieldMappingOn, 
            bool isMagFieldMapped, Int32 magFieldMappingProgress, SensitivityAcc accSensitivity, 
            SensitivityGyr gyroSensitivity, bool autocalOn, SamplingRate samplingRate, Int32 dataBuffering, float connectionInterval)
        {
            IsOffsetCompensationOn = isOffsetCompensationOn;
            IsOffsetCompensated = isOffsetCompensated;
            IsMagFieldMappingOn = isMagFieldMappingOn;
            IsMagFieldMapped = isMagFieldMapped;
            MagFieldMappingProgress = magFieldMappingProgress;
            AccSensitivity = accSensitivity;
            GyroSensitivity = gyroSensitivity;
            IsAutoCalibrationOn = autocalOn;
            SamplingRate = samplingRate;
            DataBuffering = dataBuffering;
            ConnectionInterval = connectionInterval;
        }

        public bool IsOffsetCompensationOn { get; }
        public bool IsOffsetCompensated { get; }
        public bool IsMagFieldMappingOn { get; }
        public bool IsMagFieldMapped { get; }
        public Int32 MagFieldMappingProgress { get; }
        public MagFieldMapQuality MagFieldMapQuality { get; }
        public SensitivityAcc AccSensitivity { get; }
        public SensitivityGyr GyroSensitivity { get; }
        public bool IsAutoCalibrationOn { get; }
        public SamplingRate SamplingRate { get; }
        public Int32 DataBuffering { get; }
        public float ConnectionInterval { get; }
    }

    /// <summary>
    /// Provides data for the DataPacketReadyEvent
    /// </summary>
    public class StreamPacketReceivedEventArgs : EventArgs
    {
        internal StreamPacketReceivedEventArgs(DataMode datamode, int buffering, string serialNumber, List<float[]> acc, List<float[]> gyro, List<float[]> mag,
            List<Quaternion> quaternion, float[] freeAcc, MagInterference interference, UInt32 seconds, float milliseconds, byte battery, byte marker)
        {
            DataMode = datamode;
            Buffering = buffering;  
            SerialNumber = serialNumber;
            Acc = acc;
            Gyro = gyro;
            Mag = mag;
            Quaternion = quaternion;
            FreeAcc = freeAcc;
            Interference = interference;
            Battery = battery;
            Marker = marker;
            Timestamp =  new DateTime(1970, 1, 1).AddSeconds(seconds).AddMilliseconds(milliseconds);
        }

        public DataMode DataMode { get; }
        public int Buffering { get; }
        public string SerialNumber { get; }
        public List<float[]> Acc { get; } = new List<float[]>() { new float[0] };
        public List<float[]> Gyro { get; } = new List<float[]>() { new float[0] };
        public List<float[]> Mag { get; } = new List<float[]>() { new float[0] };
        public List<Quaternion> Quaternion { get; }
        public float[] FreeAcc { get; } = new float[3];
        public MagInterference Interference { get; }
        public DateTime Timestamp { get; }
        public byte Battery { get; }
        public byte Marker { get; }
    }

}

File: Swift SDK/QSenseSwift/QSenseEventArgs.swift
//
//  QSenseEventArgs.swift
//  QSenseSwift
//
//  Created by 2M Engineering ltd on 18/07/2024.
//

import Foundation
import simd

/// <summary>
/// Provides data for the BatteryReceivedEvent
/// </summary>
public class BatteryReceivedEventArgs
{
    internal init(battery : Float)
    {
        Battery = battery;
    }

    public var Battery : Float;
}


/// <summary>
/// Provides data for the DeviceNameChangedEvent
/// </summary>
public class DeviceNameChangedEventArgs
{
    internal init(name : String)
    {
        Name = name;
    }

    public var Name : String;
}

/// <summary>
/// Provides data for the EnergyReceivedEvent
/// </summary>
public class MotionLevelReceivedEventArgs
{
    internal init(motionLevel : Float)
    {
        MotionLevel = motionLevel;
    }

    public var MotionLevel : Float;
}

/// <summary>
/// Provides data for the StateReceivedEvent
/// </summary>
public class StateReceivedEventArgs
{
    internal init(isOffsetCompensationOn : Bool, isOffsetCompensated : Bool, isMagFieldMappingOn : Bool, isMagFieldMapped : Bool, magFieldMappingProgress : Int32, accSensitivity : SensitivityAcc, gyroSensitivity : SensitivityGyr, isAutoCalOn: Bool, samplingRate : SamplingRates, dataBuffering : Int32, connectionInterval : Float)
    {
        IsOffsetCompensationOn = isOffsetCompensationOn;
        IsOffsetCompensated = isOffsetCompensated;
        IsMagFieldMappingOn = isMagFieldMappingOn;
        IsMagFieldMapped = isMagFieldMapped;
        MagFieldMappingProgress = magFieldMappingProgress;
        AccSensitivity = accSensitivity;
        GyroSensitivity = gyroSensitivity;
        IsAutoCalibrationOn = isAutoCalOn;
        SamplingRate = samplingRate;
        DataBuffering = dataBuffering;
        ConnectionInterval = connectionInterval;
    }

    public var IsOffsetCompensationOn : Bool;
    public var IsOffsetCompensated : Bool;
    public var IsMagFieldMappingOn : Bool;
    public var IsMagFieldMapped : Bool;
    public var MagFieldMappingProgress : Int32;
    public var AccSensitivity : SensitivityAcc;
    public var GyroSensitivity : SensitivityGyr;
    public var IsAutoCalibrationOn : Bool;
    public var SamplingRate : SamplingRates;
    public var DataBuffering : Int32;
    public var ConnectionInterval : Float;
}

/// <summary>
/// Provides data for the DataPacketReadyEvent
/// </summary>
public class StreamPacketReceivedEventArgs
{
    internal init(dataMode : DataModes, deviceAddress : UInt64, acc : [[Float]], gyro : [[Float]], mag : [[Float]],
                quaternion : [simd_quatf], freeAcc : [Float], time : Date, battery : UInt8, annotation : UInt8,
                  syncOk : Bool, interference : MagInterference, buffering : UInt8)
    {
        DataMode = dataMode;
        DeviceAddress = deviceAddress;
        Acc = acc;
        Gyro = gyro;
        Mag = mag;
        Quaternion = quaternion;
        FreeAcc = freeAcc;
        Time = time;
        Battery = battery;
        Annotation = annotation;
        SyncOk = syncOk;
        Interference = interference;
        Buffering = buffering;
    }

    public var DeviceAddress : UInt64;
    public var Acc : [[Float]];
    public var Gyro : [[Float]];
    public var Mag : [[Float]];
    public var Quaternion : [simd_quatf];
    public var FreeAcc : [Float];
    public var Time : Date;
    public var Battery : UInt8;
    public var Annotation : UInt8;
    public var SyncOk : Bool;
    public var DataMode : DataModes;
    public var Interference : MagInterference;
    public var Buffering : UInt8;
}
